k8s/src/test/scala/io/buoyant/k8s/IngressCacheTest.scala

```diff
@@ -406,6 +465,19 @@ class IngressCacheTest extends FunSuite with Awaits {
     assert(await(cache.matchPath(host, "/unknown-path")) == None)
   }
 
+  test("deleting an ingress does not reset the ingress cache") {
+    val service = mkWatchedIngressApiServiceReturning(
+      ingressResourceListWithMoreThanOneLinkerdIngresses,
+      ingressDeletedFromMoreThanOneLinkerdIngresses
+    )
+    val cache = new IngressCache(None, service, annotationClass, true)
+    // The ingress that was deleted is gone.
+    assert(await(cache.matchPath(host, "/linkerd-1")) == None)
+
+    // Untouched ingresses remain.
+    assert(await(cache.matchPath(host, "/linkerd-2")).get.svc == "echo2")
```

Comment:
Very minor, so feel free to take it or leave it, but I would probably try to find a way to phrase this without the `.get` call, maybe
```scala
assert(await(cache.matchPath(host, "/linkerd-2")).contains(_.svc == "echo2"))
```
or importing `org.scalatest.OptionValues` and using 
```scala
assert(await(cache.matchPath(host, "/linkerd-2")).value.svc == "echo2")
```

Reply:
No luck with the first method:

```scala
[error] /Users/negz/control/linkerd/k8s/src/test/scala/io/buoyant/k8s/IngressCacheTest.scala:478: missing parameter type for expanded function ((x$1: <error>) => x$1.s
vc.$eq$eq("echo2"))                                                  
[error]     assert(await(cache.matchPath(host, "/linkerd-2")).contains(_.svc == "echo2"))  
```

So I tried...
```scala
[error] /Users/negz/control/linkerd/k8s/src/test/scala/io/buoyant/k8s/IngressCacheTest.scala:478: missing parameter type
[error]     assert(await(cache.matchPath(host, "/linkerd-2")).contains(p => p.svc == "echo2"))
```

So I tried...
```scala
assert(await(cache.matchPath(host, "/linkerd-2")).contains((p: IngressPath) => p.svc == "echo2"))
```

And got...
```
[info] - deleting an ingress does not reset the ingress cache *** FAILED ***
[info]   Some(IngressPath(None,Some(/linkerd-2),default,echo2,7070)) did not contain io.buoyant.k8s.IngressCacheTest$$Lambda$1620/1333133483@4c5d52cd (IngressCacheTest.scala:478)
```

The `org.scalatest.OptionValues` worked though! I've switched over all the tests.